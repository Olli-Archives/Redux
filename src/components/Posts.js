import React from 'react';
import Post from './Post';
import PropTypes from 'prop-types';

function Posts({ postsArray }){

  const posts = postsArray.map((post, i)=>{
    console.log('post', post);
    return <li key={i}><Post title={post.title}/></li>;
  });

  return (
    <ul>{posts}</ul>
  );
}

Posts.propTypes = {
  postsArray: PropTypes.array.isRequired
};

export default Posts;
