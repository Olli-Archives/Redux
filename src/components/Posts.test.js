import React from 'react';
import { shallow } from 'enzyme';
import { Posts } from './Posts';


describe('Posts component', ()=>{
  it('renders list of posts', ()=>{
    const postsArray = [
      { title:'1st post', body:'1st body' },
      { title:'2nd post', body:'2nd body' },
      { title:'3rd post', body:'3rd body' },
    ];
    const wrapper = shallow(<Posts postsArray={postsArray}/>);
    expect(wrapper).toMatchSnapshot();
  });
})
;
