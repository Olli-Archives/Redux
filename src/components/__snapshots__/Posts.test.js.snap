// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts component renders list of posts 1`] = `
<ul>
  <li
    key="0"
  >
    <Post
      title="1st post"
    />
  </li>
  <li
    key="1"
  >
    <Post
      title="2nd post"
    />
  </li>
  <li
    key="2"
  >
    <Post
      title="3rd post"
    />
  </li>
</ul>
`;
